### DISCUSSION TITLE
set_body_state in PeleC in Advance

### DISCUSSION THREAD

**Discussion by jagmohas** [AMReX-Combustion/PeleC#628] [Q&A] [answered]
Hi,
I see that in advance.cpp at the end we call set_body_state. Body_state is filled in initEB.cpp and appears to be filled in with the values with state variables . May I ask how it is used and why it is needed.

**Comment by drummerdoc** [comment, marked as answer]
The "body state" is a sample of a complete state that is safely computable so that we can avoid skipping over covered cells in the lower parts of the codes called during "advance". However, when you "advance" the body state, you change it and there's a chance these covered state will eventually drift into something unsafe.  So, after each advance, we just reset it back to something safe.  It's just a hack to simplify the coding, and allows in many places the same code for boxes containing the EB as those without.

**Reply by jagmohas** [reply]
Thanks Marc for explaining. It makes sense.
